test/integration/force_torque.cc

```diff
@@ -0,0 +1,310 @@
+
+  // Create SDF describing a a force torque sensor
+  const std::string name = "TestForceTorqueSensor";
+  const std::string topic = "/ignition/sensors/test/force_torque";
+  const double updateRate = 30;
+  const bool alwaysOn = 1;
+  const bool visualize = 1;
+
+  const math::Vector3d forceNoiseMean{0.1, 0.2, 0.3};
+  // Set the stddev to 0s so we can test the mean values
+  const math::Vector3d forceNoiseStddev{0.0, 0.0, 0.0};
+
+  const math::Vector3d torqueNoiseMean{0.5, 0.6, 0.7};
+  // Set the stddev to 0s so we can test the mean values
+  const math::Vector3d torqueNoiseStddev{0.0, 0.0, 0.0};
+
+  // Create sensor SDF
+  math::Pose3d sensorPose{math::Vector3d(0.25, 0.0, 0.5),
+                          math::Quaterniond::Identity};
+  sdf::ElementPtr forcetorqueSdf;
+
+  CreateForceTorqueToSdf(name, sensorPose, updateRate, topic, alwaysOn,
+                         visualize, frame, measureDirection, forceNoiseMean,
+                         forceNoiseStddev, torqueNoiseMean, torqueNoiseStddev,
+                         forcetorqueSdf);
+
+  ASSERT_NE(nullptr, forcetorqueSdf);
+
+  // create the sensor using sensor factory
+  // try creating without specifying the sensor type and then cast it
+  ignition::sensors::SensorFactory sf;
+  auto sensor =
+      sf.CreateSensor<ignition::sensors::ForceTorqueSensor>(forcetorqueSdf);
+
+  ASSERT_NE(nullptr, sensor);
+
+  // verify initial readings
+  EXPECT_EQ(math::Vector3d::Zero, sensor->Force());
+  EXPECT_EQ(math::Vector3d::Zero, sensor->Torque());
+  EXPECT_EQ(math::Quaterniond::Identity, sensor->RotationParentInSensor());
+  EXPECT_EQ(math::Quaterniond::Identity, sensor->RotationChildInSensor());
+
+  // set state and verify readings
+  math::Vector3d force{0, 0, 1};
+  sensor->SetForce(force);
+  EXPECT_EQ(force, sensor->Force());
+
+  math::Vector3d torque{0, 1, 0};
+  sensor->SetTorque(torque);
+  EXPECT_EQ(torque, sensor->Torque());
+
+  // verify msg received on the topic
+  WaitForMessageTestHelper<ignition::msgs::Wrench> msgHelper(topic);
+  auto dt = std::chrono::steady_clock::duration(std::chrono::seconds(1));
+
+  // Set rotation of child
+  const math::Quaterniond rotChildInSensor{
+      math::Vector3d{IGN_PI_4, IGN_PI_2, 0}};
+  const math::Quaterniond rotParentInSensor{
+      math::Vector3d{0, IGN_PI_4, IGN_PI_4}};
+
+  sensor->SetRotationChildInSensor(rotChildInSensor);
+  EXPECT_EQ(rotChildInSensor, sensor->RotationChildInSensor());
+  sensor->SetRotationParentInSensor(rotParentInSensor);
+  EXPECT_EQ(rotParentInSensor, sensor->RotationParentInSensor());
+
+  sensor->Update(dt);
+  EXPECT_TRUE(msgHelper.WaitForMessage()) << msgHelper;
+  auto msg = msgHelper.Message();
+  EXPECT_EQ(1, msg.header().stamp().sec());
+  EXPECT_EQ(0, msg.header().stamp().nsec());
+
+  if (measureDirection == "parent_to_child")
+  {
+    if (frame == "child")
+    {
+      EXPECT_EQ((rotChildInSensor.Inverse() * force) + forceNoiseMean,
+                ignition::msgs::Convert(msg.force()));
```

Comment:
I personally prefer to get to the expectations on a test in a different way from the source code's implementation. For example, I think this test doesn't really help us find problems with the implementation:

```
// Code
int sum(int _a, int _b)
{
  return _a * _b;
}

// Test
EXPECT_EQ(a * b, sum(a, b));
```

A better test would be:

```
// Values obtained with a calculator
EXPECT_EQ(3, sum(1, 2));
```

I don't think the current test needs to be changed. Just a general comment.